sklearn/model_selection/_validation.py

```diff
@@ -276,6 +295,14 @@ def cross_validate(estimator, X, y=None, *, groups=None, scoring=None, cv=None,
             key = 'train_%s' % name
             ret[key] = train_scores_dict[name]
 
+    if return_predictions:
+        if return_predictions == "predict":
+            predictions = np.hstack(results["predictions"])
+        else:
+            predictions = np.vstack(results["predictions"])
+        test_indices = np.hstack(results["test_indices"])
+        ret['predictions'] = predictions[test_indices]
```

Comment:
Should this not be indexed by the argsort of test_indices? 
If the predictions and test_indices already have the same order, the indices necessary to sort test_indices would also sort the predictions.

Reply:
`cross_val_predict()` uses:

```
    inv_test_indices = np.empty(len(test_indices), dtype=int)
    inv_test_indices[test_indices] = np.arange(len(test_indices))
```

which seems to be the same as `test_indices.argsort()`